examples_x035/usbpd_sink/usbpd.h

```diff
@@ -0,0 +1,921 @@
+	uint32_t OperatingPowerIn250mW : 10u;
+	uint32_t Reserved_20_21bit : 2u; // shall be set to zero
+	uint32_t ERPCapable : 1u;
+	uint32_t UnchunkedExtendedMessage : 1u;
+	uint32_t NoUSBSuspended : 1u;
+	uint32_t USBComsCapable : 1u;
+	uint32_t CapabilityMissmatch : 1u;
+	uint32_t Giveback : 1u; // Deprecated, shall be set to zero
+	uint32_t ObjectPosition : 4u; // Reserved and shall not be used
+} USBPD_BatteryRDO_t;
+
+typedef struct
+{
+	uint32_t OperatingCurrentIn50mA : 7u;
+	uint32_t Reserved_7_8bit : 2u; // shall be set to zero
+	uint32_t OutputVoltageIn20mV : 12u;
+	uint32_t Reserved_21bit : 1u; // shall be set to zero
+	uint32_t ERPCapable : 1u;
+	uint32_t UnchunkedExtendedMessage : 1u;
+	uint32_t NoUSBSuspended : 1u;
+	uint32_t USBComsCapable : 1u;
+	uint32_t CapabilityMissmatch : 1u;
+	uint32_t Reserved_27bit : 1u; // Deprecated, shall be set to zero
+	uint32_t ObjectPosition : 4u; // Reserved and shall not be used
+} USBPD_PPS_RDO_t;
+
+typedef struct
+{
+	uint32_t OperatingCurrentIn50mA : 7u;
+	uint32_t Reserved_7_8bit : 2u; // shall be set to zero
+	uint32_t OutputVoltageIn100mV : 12u; // NOTE: Output voltage in 25mV units, the least two significant bits Shall
+	                                     // be set to zero making the effective voltage step size 100mV.
+	uint32_t Reserved_21bit : 1u; // shall be set to zero
+	uint32_t ERPCapable : 1u;
+	uint32_t UnchunkedExtendedMessage : 1u;
+	uint32_t NoUSBSuspended : 1u;
+	uint32_t USBComsCapable : 1u;
+	uint32_t CapabilityMissmatch : 1u;
+	uint32_t Reserved_27bit : 1u; // Deprecated, shall be set to zero
+	uint32_t ObjectPosition : 4u; // Reserved and shall not be used
+} USBPD_AVS_RDO_t;
+
+typedef union
+{
+	USBPD_FixedAndVariableRDO_t FixedAndVariable;
+	USBPD_BatteryRDO_t Battery;
+	USBPD_PPS_RDO_t PPS;
+} USBPD_RequestDataObject_t;
+
+static_assert( sizeof( USBPD_RequestDataObject_t ) == sizeof( uint32_t ), "USBPD_RequestDataObject_t size mismatch" );
+
+// TODO: Define the rest of the message types sections 6.4.3 -> 6.5.16
+
+typedef enum
+{
+	eUSBPD_MAX_EXTENDED_MSG_LEN = 260,
+	eUSBPD_MAX_EXTENDED_MSG_CHUNK_LEN = 26,
+	eUSBPD_MAX_EXTENDED_MSG_LEGACY_LEN = 26,
+} USBPD_ValueParameters_t;
+
+// Module public type definitions
+// -----------------------------------------------------------------------------
+
+typedef enum
+{
+	eUSBPD_OK = 0,
+	eUSBPD_BUSY,
+	eUSBPD_ERROR,
+	eUSBPD_ERROR_ARGS,
+	eUSBPD_ERROR_NOT_SUPPORTED,
+	eUSBPD_ERROR_TIMEOUT,
+} USBPD_Result_e;
+
+typedef enum
+{
+	eUSBPD_VCC_3V3 = 0,
+	eUSBPD_VCC_5V0 = 1,
+} USBPD_VCC_e;
+
+typedef enum
+{
+	eUSBPD_CCNONE = 0,
+	eUSBPD_CC1 = 1,
+	eUSBPD_CC2 = 2,
+} USBPD_CC_e;
+
+typedef enum
+{
+	eSTATE_IDLE,
+	eSTATE_CABLE_DETECT,
+	eSTATE_SOURCE_CAP,
+	eSTATE_WAIT_ACCEPT,
+	eSTATE_WAIT_PS_RDY,
+	eSTATE_PS_RDY,
+	eSTATE_MAX,
+} USBPD_State_e;
+
+/**
+ * @brief  Initialize the USB PD module
+ * @param  vcc: VCC voltage level (3.3V or 5V)
+ * @return USBPD_Result_e
+ */
+USBPD_Result_e USBPD_Init( USBPD_VCC_e vcc );
```

Comment:
1. I am really happy with the way you set all this up.
2. I really do wish there was less boiler plate above this. I think it's important for users to get to the meat sooner than later.

Reply:
This also could be resolved by having a multi-line comment at the top of the file that explains how to use it.